hostpwrctl.cpp

```diff
@@ -0,0 +1,414 @@
+#include <sdeventplus/utility/timer.hpp>
+
+#include <cstdio>
+
+constexpr auto clockId = sdeventplus::ClockId::RealTime;
+using Timer = sdeventplus::utility::Timer<clockId>;
+
+static sdbusplus::bus::bus systemBus = sdbusplus::bus::new_default();
+static sdeventplus::Event systemEvent = sdeventplus::Event::get_default();
+
+constexpr auto chassisPath = "/xyz/openbmc_project/state/chassis0";
+constexpr auto chassisIface = "xyz.openbmc_project.State.Chassis";
+constexpr auto chassisState = "CurrentPowerState";
+constexpr auto chassisStateOn =
+    "xyz.openbmc_project.State.Chassis.PowerState.On";
+constexpr auto chassisStateOff =
+    "xyz.openbmc_project.State.Chassis.PowerState.Off";
+constexpr auto chassisTransition = "RequestedPowerTransition";
+constexpr auto chassisTransitionOff =
+    "xyz.openbmc_project.State.Chassis.Transition.Off";
+
+constexpr auto hostPath = "/xyz/openbmc_project/state/host0";
+constexpr auto hostIface = "xyz.openbmc_project.State.Host";
+constexpr auto hostState = "CurrentHostState";
+constexpr auto hostStateOn = "xyz.openbmc_project.State.Host.HostState.Running";
+constexpr auto hostStateOff = "xyz.openbmc_project.State.Host.HostState.Off";
+constexpr auto hostTransition = "RequestedHostTransition";
+constexpr auto hostTransitionOn =
+    "xyz.openbmc_project.State.Host.Transition.On";
+constexpr auto hostTransitionOff =
+    "xyz.openbmc_project.State.Host.Transition.Off";
+constexpr auto hostTransitionReboot =
+    "xyz.openbmc_project.State.Host.Transition.Reboot";
+
+static std::string currentChassisState, expectedChassisState;
+static std::string currentHostState, expectedHostState;
+
+/**
+ * @brief Remove class name form the property value
+ *        For example 'xyz.foo.bar.value' -> 'value'
+ *
+ * @param value - Original value
+ *
+ * @return trimmed value
+ */
+inline std::string trimClassName(const std::string& value)
+{
+    auto last = value.rfind('.');
+    if (last && last != std::string::npos)
+    {
+        return value.substr(last + 1);
+    }
+    return value;
+}
+
+/**
+ * @brief Get D-Bus service name
+ *
+ * @param path  - object path
+ * @param iface - D-Bus interface
+ *
+ * @return D-Bus service name
+ */
+static std::string getService(const std::string& path, const std::string& iface)
+{
+    auto method = systemBus.new_method_call(
+        "xyz.openbmc_project.ObjectMapper",
+        "/xyz/openbmc_project/object_mapper",
+        "xyz.openbmc_project.ObjectMapper", "GetObject");
+
+    std::vector<std::string> ifaces = {iface};
+    method.append(path, ifaces);
+
+    std::map<std::string, std::vector<std::string>> services;
+    try
+    {
+        systemBus.call(method).read(services);
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        fprintf(stderr, "Error occured during the object mapper call: %s\n",
+                e.what());
+    }
+
+    if (!services.empty())
+    {
+        return std::move(services.begin()->first);
+    }
+
+    return "";
```

Comment:
You don't need an empty static string variable, just use `return std::string();`

Reply:
done